File: c/sort/insertSort.c
#include <stdio.h>

//插入排序

void insertSort(int  arr[],int size){
    int i,j;

    for(i=1;i<size;i++){
        int value=arr[i];//把i位置挖出来空着
        //下面需要在[0,i]中找位置把value插进去
        for(j=i-1;j>=0;j--){
            if(arr[j]>value){
                arr[j+1]=arr[j];
            }else{
                break;
            }
        }
        arr[j+1]=value;
    }

}

int main(){
    int arr[]={3,2,1,5,7,-1,5,7,9,0};
    int len= sizeof(arr)/ sizeof(int);
    insertSort(arr,len);
    int i=0;
    for(;i<len;i++){
        printf("%d ",arr[i]);
    }
    printf("\n");
    return 0;
}

File: leetcode/go/algo/sort/insertSort.go
package sort

import "algo_practice/algo"

func InsertSort(arr []int) {
	cnt := len(arr)
	for i := 1; i < cnt; i++ {
		j := i - 1;
		for j >= 0 {
			if arr[j+1] < arr[j] {
				algo.Swap(arr, j+1, j)
				j--
			} else {
				break
			}
		}
	}
}
func InsertSort1(arr []int) {
	cnt := len(arr)
	for i := 1; i < cnt; i++ {
		j := i - 1;
		tmpi := arr[j+1]
		for j >= 0 {
			//j后挪，放在j+1
			if arr[j] > tmpi {
				arr[j+1] = arr[j]
				j--
			} else {
				break
			}
		}
		arr[j+1] = tmpi
	}
}
